diff -Naur xyce-9999_or/cmake/CPackConfig.cmake xyce-9999/cmake/CPackConfig.cmake
--- xyce-9999_or/cmake/CPackConfig.cmake        2023-05-15 12:38:38.177693416 +0200
+++ xyce-9999/cmake/CPackConfig.cmake   2023-05-15 12:39:49.513921635 +0200
@@ -1,7 +1,7 @@
 ## Installation and CPack
 # copyright, readme, license, etc.
 install ( FILES ${Xyce_SOURCE_DIR}/distribution/README.TXT
-            DESTINATION doc
+            DESTINATION ${CMAKE_INSTALL_DOCDIR}
             OPTIONAL )
 
 
diff -Naur xyce-9999_or/cmake/trilinos/AMD/CMakeLists.txt xyce-9999/cmake/trilinos/AMD/CMakeLists.txt
--- xyce-9999_or/cmake/trilinos/AMD/CMakeLists.txt      2023-05-15 12:38:38.177693416 +0200
+++ xyce-9999/cmake/trilinos/AMD/CMakeLists.txt 2023-05-15 12:43:28.645636710 +0200
@@ -60,7 +60,7 @@
 
 set_property(TARGET amd PROPERTY POSITION_INDEPENDENT_CODE ON)
 
-install(TARGETS amd DESTINATION lib)
+install(TARGETS amd DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES
      ${SuiteSparsePath}/AMD/Include/amd.h
      ${SuiteSparsePath}/SuiteSparse_config/SuiteSparse_config.h
diff -Naur xyce-9999_or/src/CMakeLists.txt xyce-9999/src/CMakeLists.txt
--- xyce-9999_or/src/CMakeLists.txt     2023-05-15 12:38:38.204359928 +0200
+++ xyce-9999/src/CMakeLists.txt        2023-05-15 12:41:52.479724497 +0200
@@ -167,7 +167,7 @@
 # using different logical target names the two targets will remove each other's
 # files. This can be prevented by setting the CLEAN_DIRECT_OUTPUT property to 1."
 
-install(TARGETS XyceLib DESTINATION lib EXPORT XyceLibTarget)
+install(TARGETS XyceLib DESTINATION ${CMAKE_INSTALL_LIBDIR} EXPORT XyceLibTarget)
 install(TARGETS Xyce DESTINATION bin EXPORT XyceTarget)
 
 # Add header files needed by other codes to link against libxyce.*

diff -Naur xyce-9999_or/utils/SimulinkInterface/CMakeLists.txt xyce-9999/utils/SimulinkInterface/CMakeLists.txt
--- xyce-9999_or/utils/SimulinkInterface/CMakeLists.txt 2023-05-15 12:38:38.211026556 +0200
+++ xyce-9999/utils/SimulinkInterface/CMakeLists.txt    2023-05-15 12:43:45.132168085 +0200
@@ -5,7 +5,7 @@
   target_include_directories(xyce_sfunction PUBLIC
       ${CMAKE_CURRENT_SOURCE_DIR}
       ${Matlab_INCLUDE_DIRS} "${Matlab_ROOT_DIR}/simulink/include")
-  install(TARGETS xyce_sfunction DESTINATION lib)
+  install(TARGETS xyce_sfunction DESTINATION ${CMAKE_INSTALL_LIBDIR})
   install(FILES XyceBlocks.slx DESTINATION share/simulink)
   
   matlab_add_mex( NAME xyce2nLv_sfunction MODULE  SRC xyce2nLv_sfunction.cpp LINK_TO XyceLib )

diff -Naur xyce-9999_or/utils/XyceCInterface/CMakeLists.txt xyce-9999/utils/XyceCInterface/CMakeLists.txt
--- xyce-9999_or/utils/XyceCInterface/CMakeLists.txt    2023-05-15 12:38:38.211026556 +0200
+++ xyce-9999/utils/XyceCInterface/CMakeLists.txt       2023-05-15 12:43:08.092469721 +0200
@@ -21,7 +21,7 @@
 endif()
 
 
-install( TARGETS xycecinterface DESTINATION lib OPTIONAL)
+install( TARGETS xycecinterface DESTINATION ${CMAKE_INSTALL_LIBDIR} OPTIONAL)
 configure_file(xyce_interface.py.cmake xyce_interface.py @ONLY )
 install( FILES N_CIR_XyceCInterface.h DESTINATION include OPTIONAL)
 install( FILES ${CMAKE_CURRENT_BINARY_DIR}/xyce_interface.py DESTINATION share OPTIONAL)

diff -Naur xyce-9999_or/src/CMakeLists.txt xyce-9999/src/CMakeLists.txt
--- xyce-9999_or/src/CMakeLists.txt     2023-05-26 18:32:42.704669936 +0200
+++ xyce-9999/src/CMakeLists.txt        2023-05-26 18:37:51.203869009 +0200
@@ -318,11 +318,11 @@
 configure_package_config_file(
   cmake/XyceConfig.cmake.in
   ${CMAKE_CURRENT_BINARY_DIR}/cmake/XyceConfig.cmake
-  INSTALL_DESTINATION lib/cmake
+  INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake
   PATH_VARS Xyce_Trilinos_DIR
 )
 
-set(ConfigPackageLocation lib/cmake/Xyce)
+set(ConfigPackageLocation ${CMAKE_INSTALL_LIBDIR}/cmake/Xyce)
 install(EXPORT XyceLibTarget
   FILE
     XyceTargets.cmake
