diff -Naur or-tools-9999_or/ortools/sat/subsolver.cc or-tools-9999/ortools/sat/subsolver.cc
--- or-tools-9999_or/ortools/sat/subsolver.cc   2023-10-23 23:50:33.232897043 +0200
+++ or-tools-9999/ortools/sat/subsolver.cc      2023-10-23 23:50:54.142977895 +0200
@@ -266,7 +266,7 @@
       task();
 
       const absl::MutexLock mutex_lock(&mutex);
-      DCHECK_NE(subsolvers[best], nullptr);
+      //DCHECK_NE(subsolvers[best], nullptr);
       DCHECK_GT(num_in_flight_per_subsolvers[best], 0);
       num_in_flight_per_subsolvers[best]--;
       VLOG(1) << name << " done in " << timer.Get() << "s.";
