diff -Naur Trilinos-trilinos-release-13-4-1_or/packages/teuchos/core/src/Teuchos_BigUIntDecl.hpp Trilinos-trilinos-release-13-4-1/packages/teuchos/core/src/Teuchos_BigUIntDecl.hpp
--- Trilinos-trilinos-release-13-4-1_or/packages/teuchos/core/src/Teuchos_BigUIntDecl.hpp       2023-05-13 23:56:33.699073588 +0200
+++ Trilinos-trilinos-release-13-4-1/packages/teuchos/core/src/Teuchos_BigUIntDecl.hpp  2023-05-13 23:56:59.079010911 +0200
@@ -43,6 +43,7 @@
 #define TEUCHOS_BIG_UINT_DECL_HPP

 #include <iosfwd>
+#include <cstdint>

 /*! \file Teuchos_BigUIntDecl.hpp
     \brief Arbitrary-precision unsigned integer declaration.
