diff -Naur Trilinos-trilinos-release-13-4-1_or/packages/kokkos/core/src/impl/Kokkos_MemoryPool.cpp Trilinos-trilinos-release-13-4-1/packages/kokkos/core/src/impl/Kokkos_MemoryPool.cpp
--- Trilinos-trilinos-release-13-4-1_or/packages/kokkos/core/src/impl/Kokkos_MemoryPool.cpp     2023-06-11 21:27:20.637681333 +0200
+++ Trilinos-trilinos-release-13-4-1/packages/kokkos/core/src/impl/Kokkos_MemoryPool.cpp        2023-06-11 21:28:23.316933920 +0200
@@ -46,6 +46,7 @@
 
 #include <ostream>
 #include <sstream>
+#include <cstdint>
 
 //----------------------------------------------------------------------------
 //----------------------------------------------------------------------------
diff -Naur Trilinos-trilinos-release-13-4-1_or/packages/stk/stk_util/stk_util/ngp/NgpSpaces.hpp Trilinos-trilinos-release-13-4-1/packages/stk/stk_util/stk_util/ngp/NgpSpaces.hpp
--- Trilinos-trilinos-release-13-4-1_or/packages/stk/stk_util/stk_util/ngp/NgpSpaces.hpp        2023-06-12 22:13:54.653058491 +0200
+++ Trilinos-trilinos-release-13-4-1/packages/stk/stk_util/stk_util/ngp/NgpSpaces.hpp   2023-06-12 22:22:22.234504055 +0200
@@ -49,6 +49,8 @@
 using ExecSpace = Kokkos::Experimental::HIP;
 #elif defined(KOKKOS_ENABLE_OPENMP)
 using ExecSpace = Kokkos::OpenMP;
+#elif defined(KOKKOS_ENABLE_THREADS)
+using ExecSpace = Kokkos::Threads;
 #else
 using ExecSpace = Kokkos::Serial;
 #endif
@@ -59,6 +61,8 @@
 using HostExecSpace = Kokkos::Serial;
 #elif defined(KOKKOS_ENABLE_OPENMP)
 using HostExecSpace = Kokkos::OpenMP;
+#elif defined(KOKKOS_ENABLE_THREADS)
+using HostExecSpace = Kokkos::Threads;
 #else
 using HostExecSpace = Kokkos::Serial;
 #endif
@@ -69,6 +73,8 @@
 using MemSpace = Kokkos::Experimental::HIPSpace;
 #elif defined(KOKKOS_ENABLE_OPENMP)
 using MemSpace = Kokkos::OpenMP;
+#elif defined(KOKKOS_ENABLE_THREADS)
+using MemSpace = Kokkos::Threads;
 #else
 using MemSpace = Kokkos::HostSpace;
 #endif
@@ -83,6 +89,8 @@
 using UVMMemSpace = Kokkos::Experimental::HIPHostPinnedSpace;
 #elif defined(KOKKOS_ENABLE_OPENMP)
 using UVMMemSpace = Kokkos::OpenMP;
+#elif defined(KOKKOS_ENABLE_THREADS)
+using UVMMemSpace = Kokkos::Threads;
 #else
 using UVMMemSpace = Kokkos::HostSpace;
 #endif
