diff -Naur Trilinos-trilinos-release-14-0-0_or/packages/stk/stk_util/stk_util/ngp/NgpSpaces.hpp Trilinos-trilinos-release-14-0-0/packages/stk/stk_util/stk_util/ngp/NgpSpaces.hpp
--- Trilinos-trilinos-release-14-0-0_or/packages/stk/stk_util/stk_util/ngp/NgpSpaces.hpp        2023-06-14 23:24:20.313033330 +0200
+++ Trilinos-trilinos-release-14-0-0/packages/stk/stk_util/stk_util/ngp/NgpSpaces.hpp   2023-06-14 23:31:11.551230025 +0200
@@ -39,36 +39,21 @@
 namespace stk {
 namespace ngp {
 
-#ifdef KOKKOS_ENABLE_CUDA
-using ExecSpace = Kokkos::Cuda;
-#elif defined(KOKKOS_ENABLE_HIP)
-using ExecSpace = Kokkos::Experimental::HIP;
-#elif defined(KOKKOS_ENABLE_OPENMP)
-using ExecSpace = Kokkos::OpenMP;
-#else
-using ExecSpace = Kokkos::Serial;
-#endif
+using ExecSpace = Kokkos::DefaultExecutionSpace;
 
-#ifdef KOKKOS_ENABLE_CUDA
-using HostExecSpace = Kokkos::Serial;
-#elif defined(KOKKOS_ENABLE_HIP)
-using HostExecSpace = Kokkos::Serial;
-#elif defined(KOKKOS_ENABLE_OPENMP)
-using HostExecSpace = Kokkos::OpenMP;
-#else
-using HostExecSpace = Kokkos::Serial;
-#endif
+using HostExecSpace = Kokkos::DefaultHostExecutionSpace;
 
 #ifdef KOKKOS_ENABLE_CUDA
 using MemSpace = Kokkos::CudaSpace;
 #elif defined(KOKKOS_ENABLE_HIP)
-using MemSpace = Kokkos::Experimental::HIPSpace;
-#elif defined(KOKKOS_ENABLE_OPENMP)
-using MemSpace = Kokkos::OpenMP;
+using MemSpace = Kokkos::HIPSpace;
 #else
-using MemSpace = Kokkos::HostSpace;
+using MemSpace = ExecSpace::memory_space;
 #endif
 
+#ifdef KOKKOS_HAS_SHARED_SPACE
+using UVMMemSpace = Kokkos::SharedSpace;
+#else
 #ifdef KOKKOS_ENABLE_CUDA
 #ifdef KOKKOS_ENABLE_CUDA_UVM
 using UVMMemSpace = Kokkos::CudaUVMSpace;
@@ -82,7 +67,11 @@
 #else
 using UVMMemSpace = Kokkos::HostSpace;
 #endif
+#endif
 
+#ifdef KOKKOS_HAS_SHARED_SPACE
+using HostPinnedSpace = Kokkos::SharedHostPinnedSpace;
+#else
 #ifdef KOKKOS_ENABLE_CUDA
 using HostPinnedSpace = Kokkos::CudaHostPinnedSpace;
 #elif defined(KOKKOS_ENABLE_HIP)
@@ -90,6 +79,7 @@
 #else
 using HostPinnedSpace = MemSpace;
 #endif
+#endif
 
 #ifdef KOKKOS_ENABLE_HIP
 template <typename ExecutionSpace>
