diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8dcd4cf004..77ddc2381a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -134,6 +134,23 @@ set(SLIC3R_GTK "3" CACHE STRING "GTK version to use with wxWidgets on Linux")
 
 set(IS_CROSS_COMPILE FALSE)
 
+option(SYS_LIBS "Compile OrcaSlicer with system libraries, if possible (e.g. 'opencv;openvdb;wxwidgets;boost;tbb;cgal')" ${SYS_LIBS})
+
+if (SYS_LIBS)
+    set(SYS_LIBS_ALL "opencv;openvdb;wxwidgets;boost;tbb;cgal")
+    if (SYS_LIBS MATCHES "all")
+        set(SYS_LIBS ${SYS_LIBS_ALL})
+    endif()
+    FOREACH(lib ${SYS_LIBS})
+        if (NOT lib IN_LIST SYS_LIBS_ALL)
+            message(FATAL_ERROR "Unknown library specified in SYS_LIBS: ${lib}. Valid options are: ${SYS_LIBS_ALL} or 'all'.")
+        endif()
+        string(TOUPPER ${lib} lib_upper)
+        set(SYS_LIB_${lib_upper} ON CACHE BOOL "Try to use system library for ${lib}")
+        message(STATUS "System library for ${lib}: ${SYS_LIB_${lib_upper}}")
+    ENDFOREACH()
+endif()
+
 option (COLORED_OUTPUT "Always produce ANSI-colored output (GNU/Clang only)." OFF)
 if (${COLORED_OUTPUT})
     if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
@@ -565,7 +582,7 @@ list(REMOVE_DUPLICATES CMAKE_C_FLAGS_RELWITHDEBINFO)
 list(REMOVE_DUPLICATES CMAKE_CXX_FLAGS_RELWITHDEBINFO)
 
 # Find and configure boost
-if(SLIC3R_STATIC)
+if(SLIC3R_STATIC AND NOT SYS_LIB_BOOST)
     # Use static boost libraries.
     set(Boost_USE_STATIC_LIBS ON)
     # Use boost libraries linked statically to the C++ runtime.
@@ -581,7 +598,12 @@ if(POLICY CMP0167)
   cmake_policy(SET CMP0167 NEW)
 endif()
 set(Boost_NO_SYSTEM_PATHS TRUE)
-find_package(Boost 1.83.0 REQUIRED COMPONENTS system filesystem thread log log_setup locale regex chrono atomic date_time iostreams program_options nowide)
+set(Boost_COMPONENTS filesystem thread log log_setup locale regex chrono atomic date_time iostreams program_options nowide process)
+if(NOT SYS_LIB_BOOST)
+    # Have only on static library
+    list(APPEND Boost_COMPONENTS system)
+endif()
+find_package(Boost 1.90.0 REQUIRED COMPONENTS ${Boost_COMPONENTS})
 
 # Find and configure Eigen3
 find_package(Eigen3 5.0.1 REQUIRED)
@@ -594,7 +616,7 @@ if (APPLE)
     target_compile_definitions(boost_headeronly INTERFACE BOOST_ASIO_DISABLE_KQUEUE)
 endif()
 
-if(NOT SLIC3R_STATIC)
+if(NOT SLIC3R_STATIC OR SYS_LIB_BOOST)
     target_compile_definitions(boost_headeronly INTERFACE BOOST_LOG_DYN_LINK)
 endif()
 
@@ -614,7 +636,7 @@ target_include_directories(boost_headeronly SYSTEM INTERFACE ${Boost_INCLUDE_DIR
 target_link_libraries(boost_libs INTERFACE boost_headeronly ${Boost_LIBRARIES})
 
 # Find and configure intel-tbb
-if(SLIC3R_STATIC)
+if(SLIC3R_STATIC AND NOT SYS_LIB_TBB)
     set(TBB_STATIC 1)
 endif()
 set(TBB_DEBUG 1)
@@ -751,14 +773,19 @@ endforeach()
 find_package(NLopt 1.4 REQUIRED)
 
 
-if(SLIC3R_STATIC)
+if(SLIC3R_STATIC AND NOT SYS_LIB_OPENVDB)
     set(OPENVDB_USE_STATIC_LIBS ON)
     set(USE_BLOSC TRUE)
 endif ()
 
+find_package(OpenEXR QUIET)
 find_package(OpenVDB 5.0 COMPONENTS openvdb)
 if(OpenVDB_FOUND)
-    slic3r_remap_configs(IlmBase::Half RelWithDebInfo Release)
+    if (OpenEXR_VERSION VERSION_LESS "3.0.0")
+        slic3r_remap_configs(IlmBase::Half RelWithDebInfo Release)
+    else()
+        slic3r_remap_configs(Imath RelWithDebInfo Release)
+    endif()
     slic3r_remap_configs(Blosc::blosc RelWithDebInfo Release)
 else ()
     message(FATAL_ERROR "OpenVDB could not be found with the bundled find module. "
diff --git a/cmake/modules/FindBlosc.cmake b/cmake/modules/FindBlosc.cmake
new file mode 100644
index 0000000000..e12c0c386d
--- /dev/null
+++ b/cmake/modules/FindBlosc.cmake
@@ -0,0 +1,429 @@
+# Copyright Contributors to the OpenVDB Project
+# SPDX-License-Identifier: Apache-2.0
+#
+#[=======================================================================[.rst:
+
+FindBlosc
+---------
+
+Find Blosc include dirs and libraries
+
+Use this module by invoking find_package with the form::
+
+  find_package(Blosc
+    [version] [EXACT]      # Minimum or EXACT version e.g. 1.5.0
+    [REQUIRED]             # Fail with error if Blosc is not found
+    )
+
+IMPORTED Targets
+^^^^^^^^^^^^^^^^
+
+``Blosc::blosc``
+  This module defines IMPORTED target Blosc::Blosc, if Blosc has been found.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This will define the following variables:
+
+``Blosc_FOUND``
+  True if the system has the Blosc library.
+``Blosc_VERSION``
+  The version of the Blosc library which was found.
+``Blosc_INCLUDE_DIRS``
+  Include directories needed to use Blosc.
+``Blosc_RELEASE_LIBRARIES``
+  Libraries needed to link to the release version of Blosc.
+``Blosc_RELEASE_LIBRARY_DIRS``
+  Blosc release library directories.
+``Blosc_DEBUG_LIBRARIES``
+  Libraries needed to link to the debug version of Blosc.
+``Blosc_DEBUG_LIBRARY_DIRS``
+  Blosc debug library directories.
+
+Deprecated - use [RELEASE|DEBUG] variants:
+
+``Blosc_LIBRARIES``
+  Libraries needed to link to Blosc.
+``Blosc_LIBRARY_DIRS``
+  Blosc library directories.
+
+Cache Variables
+^^^^^^^^^^^^^^^
+
+The following cache variables may also be set:
+
+``Blosc_INCLUDE_DIR``
+  The directory containing ``blosc.h``.
+``Blosc_LIBRARY``
+  The path to the Blosc library. may include target_link_libraries() debug/optimized keywords
+``Blosc_LIBRARY_RELEASE``
+  The path to the Blosc release library.
+``Blosc_LIBRARY_DEBUG``
+  The path to the Blosc debug library.
+
+Hints
+^^^^^
+
+Instead of explicitly setting the cache variables, the following variables
+may be provided to tell this module where to look.
+
+``Blosc_ROOT``
+  Preferred installation prefix.
+``BLOSC_INCLUDEDIR``
+  Preferred include directory e.g. <prefix>/include
+``BLOSC_LIBRARYDIR``
+  Preferred library directory e.g. <prefix>/lib
+``BLOSC_DEBUG_SUFFIX``
+  Suffix of the debug version of blosc. Defaults to "_d", OR the empty string for VCPKG_TOOLCHAIN
+``SYSTEM_LIBRARY_PATHS``
+  Global list of library paths intended to be searched by and find_xxx call
+``BLOSC_USE_STATIC_LIBS``
+  Only search for static blosc libraries
+``BLOSC_USE_EXTERNAL_SOURCES``
+  Set to ON if Blosc has been built using external sources for LZ4, snappy,
+  zlib and zstd. Default is OFF.
+``DISABLE_CMAKE_SEARCH_PATHS``
+  Disable CMakes default search paths for find_xxx calls in this module
+
+#]=======================================================================]
+
+cmake_minimum_required(VERSION 3.20)
+include(GNUInstallDirs)
+
+mark_as_advanced(
+  Blosc_INCLUDE_DIR
+  Blosc_LIBRARY
+)
+
+set(_FIND_BLOSC_ADDITIONAL_OPTIONS "")
+if(DISABLE_CMAKE_SEARCH_PATHS)
+  set(_FIND_BLOSC_ADDITIONAL_OPTIONS NO_DEFAULT_PATH)
+endif()
+
+# Set _BLOSC_ROOT based on a user provided root var. Xxx_ROOT and ENV{Xxx_ROOT}
+# are prioritised over the legacy capitalized XXX_ROOT variables for matching
+# CMake 3.12 behaviour
+# @todo  deprecate -D and ENV BLOSC_ROOT from CMake 3.12
+if(Blosc_ROOT)
+  set(_BLOSC_ROOT ${Blosc_ROOT})
+elseif(DEFINED ENV{Blosc_ROOT})
+  set(_BLOSC_ROOT $ENV{Blosc_ROOT})
+elseif(BLOSC_ROOT)
+  set(_BLOSC_ROOT ${BLOSC_ROOT})
+elseif(DEFINED ENV{BLOSC_ROOT})
+  set(_BLOSC_ROOT $ENV{BLOSC_ROOT})
+endif()
+
+# Additionally try and use pkconfig to find blosc
+if(USE_PKGCONFIG)
+  if(NOT DEFINED PKG_CONFIG_FOUND)
+    find_package(PkgConfig)
+  endif()
+  if(PKG_CONFIG_FOUND)
+    pkg_check_modules(PC_Blosc QUIET blosc)
+  endif()
+endif()
+
+# ------------------------------------------------------------------------
+#  Search for blosc include DIR
+# ------------------------------------------------------------------------
+
+set(_BLOSC_INCLUDE_SEARCH_DIRS "")
+list(APPEND _BLOSC_INCLUDE_SEARCH_DIRS
+  ${BLOSC_INCLUDEDIR}
+  ${_BLOSC_ROOT}
+  ${PC_Blosc_INCLUDE_DIRS}
+  ${SYSTEM_LIBRARY_PATHS}
+)
+
+# Look for a standard blosc header file.
+find_path(Blosc_INCLUDE_DIR blosc.h
+  ${_FIND_BLOSC_ADDITIONAL_OPTIONS}
+  PATHS ${_BLOSC_INCLUDE_SEARCH_DIRS}
+  PATH_SUFFIXES ${CMAKE_INSTALL_INCLUDEDIR} include
+)
+
+if(EXISTS "${Blosc_INCLUDE_DIR}/blosc.h")
+  file(STRINGS "${Blosc_INCLUDE_DIR}/blosc.h"
+    _blosc_version_major_string REGEX "#define BLOSC_VERSION_MAJOR +[0-9]+ "
+  )
+  string(REGEX REPLACE "#define BLOSC_VERSION_MAJOR +([0-9]+).*$" "\\1"
+    _blosc_version_major_string "${_blosc_version_major_string}"
+  )
+  string(STRIP "${_blosc_version_major_string}" Blosc_VERSION_MAJOR)
+
+  file(STRINGS "${Blosc_INCLUDE_DIR}/blosc.h"
+     _blosc_version_minor_string REGEX "#define BLOSC_VERSION_MINOR +[0-9]+ "
+  )
+  string(REGEX REPLACE "#define BLOSC_VERSION_MINOR +([0-9]+).*$" "\\1"
+    _blosc_version_minor_string "${_blosc_version_minor_string}"
+  )
+  string(STRIP "${_blosc_version_minor_string}" Blosc_VERSION_MINOR)
+
+  file(STRINGS "${Blosc_INCLUDE_DIR}/blosc.h"
+     _blosc_version_release_string REGEX "#define BLOSC_VERSION_RELEASE +[0-9]+ "
+  )
+  string(REGEX REPLACE "#define BLOSC_VERSION_RELEASE +([0-9]+).*$" "\\1"
+    _blosc_version_release_string "${_blosc_version_release_string}"
+  )
+  string(STRIP "${_blosc_version_release_string}" Blosc_VERSION_RELEASE)
+
+  unset(_blosc_version_major_string)
+  unset(_blosc_version_minor_string)
+  unset(_blosc_version_release_string)
+
+  set(Blosc_VERSION ${Blosc_VERSION_MAJOR}.${Blosc_VERSION_MINOR}.${Blosc_VERSION_RELEASE})
+endif()
+
+# ------------------------------------------------------------------------
+#  Search for blosc lib DIR
+# ------------------------------------------------------------------------
+
+set(_BLOSC_LIBRARYDIR_SEARCH_DIRS "")
+list(APPEND _BLOSC_LIBRARYDIR_SEARCH_DIRS
+  ${BLOSC_LIBRARYDIR}
+  ${_BLOSC_ROOT}
+  ${PC_Blosc_LIBRARY_DIRS}
+  ${SYSTEM_LIBRARY_PATHS}
+)
+
+set(Blosc_LIB_COMPONENTS "")
+# NOTE: Search for debug version first (see vcpkg hack)
+list(APPEND BLOSC_BUILD_TYPES DEBUG RELEASE)
+
+foreach(BUILD_TYPE ${BLOSC_BUILD_TYPES})
+  set(_BLOSC_LIB_NAME blosc)
+
+  set(_BLOSC_CMAKE_IGNORE_PATH ${CMAKE_IGNORE_PATH})
+  if(VCPKG_TOOLCHAIN)
+    # Blosc is installed very strangely in VCPKG (debug/release libs have the
+    # same name, static build uses external deps, dll doesn't) and blosc itself
+    # comes with almost zero downstream CMake support for us to detect settings.
+    # We should not support external package managers in our own modules like
+    # this, but there doesn't seem to be a work around
+    if(NOT DEFINED BLOSC_DEBUG_SUFFIX)
+      set(BLOSC_DEBUG_SUFFIX "")
+    endif()
+    if(BUILD_TYPE STREQUAL RELEASE)
+      if(EXISTS ${Blosc_LIBRARY_DEBUG})
+        get_filename_component(_BLOSC_DEBUG_DIR ${Blosc_LIBRARY_DEBUG} DIRECTORY)
+        list(APPEND CMAKE_IGNORE_PATH ${_BLOSC_DEBUG_DIR})
+      endif()
+    endif()
+  endif()
+
+  if(BUILD_TYPE STREQUAL DEBUG)
+    if(NOT DEFINED BLOSC_DEBUG_SUFFIX)
+      set(BLOSC_DEBUG_SUFFIX _d)
+    endif()
+    set(_BLOSC_LIB_NAME "${_BLOSC_LIB_NAME}${BLOSC_DEBUG_SUFFIX}")
+  endif()
+
+  find_library(Blosc_LIBRARY_${BUILD_TYPE} ${_BLOSC_LIB_NAME}
+    ${_FIND_BLOSC_ADDITIONAL_OPTIONS}
+    PATHS ${_BLOSC_LIBRARYDIR_SEARCH_DIRS}
+    PATH_SUFFIXES ${CMAKE_INSTALL_LIBDIR} lib64 lib
+  )
+
+  list(APPEND Blosc_LIB_COMPONENTS ${Blosc_LIBRARY_${BUILD_TYPE}})
+  set(CMAKE_IGNORE_PATH ${_BLOSC_CMAKE_IGNORE_PATH})
+endforeach()
+
+if(Blosc_LIBRARY_DEBUG AND Blosc_LIBRARY_RELEASE)
+  # if the generator is multi-config or if CMAKE_BUILD_TYPE is set for
+  # single-config generators, set optimized and debug libraries
+  get_property(_isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+  if(_isMultiConfig OR CMAKE_BUILD_TYPE)
+    set(Blosc_LIBRARY optimized ${Blosc_LIBRARY_RELEASE} debug ${Blosc_LIBRARY_DEBUG})
+  else()
+    # For single-config generators where CMAKE_BUILD_TYPE has no value,
+    # just use the release libraries
+    set(Blosc_LIBRARY ${Blosc_LIBRARY_RELEASE})
+  endif()
+  # FIXME: This probably should be set for both cases
+  set(Blosc_LIBRARIES optimized ${Blosc_LIBRARY_RELEASE} debug ${Blosc_LIBRARY_DEBUG})
+endif()
+
+# if only the release version was found, set the debug variable also to the release version
+if(Blosc_LIBRARY_RELEASE AND NOT Blosc_LIBRARY_DEBUG)
+  set(Blosc_LIBRARY_DEBUG ${Blosc_LIBRARY_RELEASE})
+  set(Blosc_LIBRARY       ${Blosc_LIBRARY_RELEASE})
+  set(Blosc_LIBRARIES     ${Blosc_LIBRARY_RELEASE})
+endif()
+
+# if only the debug version was found, set the release variable also to the debug version
+if(Blosc_LIBRARY_DEBUG AND NOT Blosc_LIBRARY_RELEASE)
+  set(Blosc_LIBRARY_RELEASE ${Blosc_LIBRARY_DEBUG})
+  set(Blosc_LIBRARY         ${Blosc_LIBRARY_DEBUG})
+  set(Blosc_LIBRARIES       ${Blosc_LIBRARY_DEBUG})
+endif()
+
+# If the debug & release library ends up being the same, omit the keywords
+if("${Blosc_LIBRARY_RELEASE}" STREQUAL "${Blosc_LIBRARY_DEBUG}")
+  set(Blosc_LIBRARY   ${Blosc_LIBRARY_RELEASE} )
+  set(Blosc_LIBRARIES ${Blosc_LIBRARY_RELEASE} )
+endif()
+
+if(Blosc_LIBRARY)
+  set(Blosc_FOUND TRUE)
+else()
+  set(Blosc_FOUND FALSE)
+endif()
+
+# ------------------------------------------------------------------------
+#  Cache and set Blosc_FOUND
+# ------------------------------------------------------------------------
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(Blosc
+  FOUND_VAR Blosc_FOUND
+  REQUIRED_VARS
+    Blosc_LIBRARY
+    Blosc_INCLUDE_DIR
+  VERSION_VAR Blosc_VERSION
+)
+
+if(NOT Blosc_FOUND)
+  if(Blosc_FIND_REQUIRED)
+    message(FATAL_ERROR "Unable to find Blosc")
+  endif()
+  return()
+endif()
+
+# Partition release/debug lib vars
+
+set(Blosc_RELEASE_LIBRARIES ${Blosc_LIBRARY_RELEASE})
+get_filename_component(Blosc_RELEASE_LIBRARY_DIRS ${Blosc_RELEASE_LIBRARIES} DIRECTORY)
+set(Blosc_DEBUG_LIBRARIES ${Blosc_LIBRARY_DEBUG})
+get_filename_component(Blosc_DEBUG_LIBRARY_DIRS ${Blosc_DEBUG_LIBRARIES} DIRECTORY)
+set(Blosc_LIBRARIES ${Blosc_RELEASE_LIBRARIES})
+set(Blosc_LIBRARY_DIRS ${Blosc_RELEASE_LIBRARY_DIRS})
+set(Blosc_INCLUDE_DIRS ${Blosc_INCLUDE_DIR})
+set(Blosc_INCLUDE_DIRS ${Blosc_INCLUDE_DIR})
+
+# Configure lib type. If XXX_USE_STATIC_LIBS, we always assume a static
+# lib is in use. If win32, we can't mark the import .libs as shared, so
+# these are always marked as UNKNOWN. Otherwise, infer from extension.
+set(BLOSC_LIB_TYPE UNKNOWN)
+if(BLOSC_USE_STATIC_LIBS)
+  set(BLOSC_LIB_TYPE STATIC)
+elseif(UNIX)
+  get_filename_component(_BLOSC_EXT ${Blosc_LIBRARY_RELEASE} EXT)
+  if(_BLOSC_EXT STREQUAL ".a")
+    set(BLOSC_LIB_TYPE STATIC)
+  elseif(_BLOSC_EXT STREQUAL ".so" OR
+         _BLOSC_EXT STREQUAL ".dylib")
+    set(BLOSC_LIB_TYPE SHARED)
+  endif()
+endif()
+
+get_filename_component(Blosc_LIBRARY_DIRS ${Blosc_LIBRARY_RELEASE} DIRECTORY)
+
+if(NOT TARGET Blosc::blosc)
+  add_library(Blosc::blosc ${BLOSC_LIB_TYPE} IMPORTED)
+  set_target_properties(Blosc::blosc PROPERTIES
+    INTERFACE_COMPILE_OPTIONS "${PC_Blosc_CFLAGS_OTHER}"
+    INTERFACE_INCLUDE_DIRECTORIES "${Blosc_INCLUDE_DIRS}")
+
+  # Standard location
+  set_target_properties(Blosc::blosc PROPERTIES
+    IMPORTED_LINK_INTERFACE_LANGUAGES "CXX"
+    IMPORTED_LOCATION "${Blosc_LIBRARY}")
+
+  # Release location
+  if(EXISTS "${Blosc_LIBRARY_RELEASE}")
+    set_property(TARGET Blosc::blosc APPEND PROPERTY
+      IMPORTED_CONFIGURATIONS RELEASE)
+    set_target_properties(Blosc::blosc PROPERTIES
+      IMPORTED_LINK_INTERFACE_LANGUAGES_RELEASE "CXX"
+      IMPORTED_LOCATION_RELEASE "${Blosc_LIBRARY_RELEASE}")
+  endif()
+
+  # Debug location
+  if(EXISTS "${Blosc_LIBRARY_DEBUG}")
+    set_property(TARGET Blosc::blosc APPEND PROPERTY
+      IMPORTED_CONFIGURATIONS DEBUG)
+    set_target_properties(Blosc::blosc PROPERTIES
+      IMPORTED_LINK_INTERFACE_LANGUAGES_DEBUG "CXX"
+      IMPORTED_LOCATION_DEBUG "${Blosc_LIBRARY_DEBUG}")
+  endif()
+
+  # Blosc may optionally be compiled with external sources for
+  # lz4, snappy and zlib . Add them as interface libs if requested
+  # (there doesn't seem to be a way to figure this out automatically).
+  # We assume they live along side blosc
+  if(BLOSC_USE_EXTERNAL_SOURCES)
+    set_target_properties(Blosc::blosc PROPERTIES
+      INTERFACE_LINK_DIRECTORIES
+         "\$<\$<CONFIG:Release>:${Blosc_RELEASE_LIBRARY_DIRS}>;\$<\$<CONFIG:Debug>:${Blosc_DEBUG_LIBRARY_DIRS}>")
+
+    set(BLOSC_EXTERNAL_LIBRARIES lz4 snappy zlib zstd)
+
+    foreach(BLOSC_EXTERNAL_LIB ${BLOSC_EXTERNAL_LIBRARIES})
+
+      foreach(BUILD_TYPE ${BLOSC_BUILD_TYPES})
+        set(_BLOSC_CMAKE_IGNORE_PATH ${CMAKE_IGNORE_PATH})
+
+        if(VCPKG_TOOLCHAIN)
+          if(BUILD_TYPE STREQUAL RELEASE)
+            list(APPEND CMAKE_IGNORE_PATH ${Blosc_DEBUG_LIBRARY_DIRS})
+          else()
+            list(APPEND CMAKE_IGNORE_PATH ${Blosc_RELEASE_LIBRARY_DIRS})
+          endif()
+        endif()
+
+        find_library(${BLOSC_EXTERNAL_LIB}_LIBRARY_${BUILD_TYPE} ${BLOSC_EXTERNAL_LIB}
+          ${_FIND_BLOSC_ADDITIONAL_OPTIONS}
+          PATHS ${_BLOSC_LIBRARYDIR_SEARCH_DIRS}
+          PATH_SUFFIXES ${CMAKE_INSTALL_LIBDIR} lib64 lib)
+
+        if(NOT ${BLOSC_EXTERNAL_LIB}_LIBRARY_${BUILD_TYPE})
+          if(BUILD_TYPE STREQUAL DEBUG)
+            find_library(${BLOSC_EXTERNAL_LIB}_LIBRARY_${BUILD_TYPE} "${BLOSC_EXTERNAL_LIB}d"
+              ${_FIND_BLOSC_ADDITIONAL_OPTIONS}
+              PATHS ${_BLOSC_LIBRARYDIR_SEARCH_DIRS}
+              PATH_SUFFIXES ${CMAKE_INSTALL_LIBDIR} lib64 lib)
+          endif()
+        endif()
+
+        set(CMAKE_IGNORE_PATH ${_BLOSC_CMAKE_IGNORE_PATH})
+      endforeach()
+
+      if(${BLOSC_EXTERNAL_LIB}_LIBRARY_DEBUG AND ${BLOSC_EXTERNAL_LIB}_LIBRARY_RELEASE)
+        # if the generator is multi-config or if CMAKE_BUILD_TYPE is set for
+        # single-config generators, set optimized and debug libraries
+        get_property(_isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+        if(_isMultiConfig OR CMAKE_BUILD_TYPE)
+          set(${BLOSC_EXTERNAL_LIB}_LIBRARY optimized ${${BLOSC_EXTERNAL_LIB}_LIBRARY_RELEASE} debug ${${BLOSC_EXTERNAL_LIB}_LIBRARY_DEBUG})
+        else()
+          # For single-config generators where CMAKE_BUILD_TYPE has no value,
+          # just use the release libraries
+          set(${BLOSC_EXTERNAL_LIB}_LIBRARY ${${BLOSC_EXTERNAL_LIB}_LIBRARY_RELEASE})
+        endif()
+        # FIXME: This probably should be set for both cases
+        set(${BLOSC_EXTERNAL_LIB}_LIBRARIES optimized ${${BLOSC_EXTERNAL_LIB}_LIBRARY_RELEASE} debug ${${BLOSC_EXTERNAL_LIB}_LIBRARY_DEBUG})
+      endif()
+
+      # if only the release version was found, set the debug variable also to the release version
+      if(${BLOSC_EXTERNAL_LIB}_LIBRARY_RELEASE AND NOT ${BLOSC_EXTERNAL_LIB}_LIBRARY_DEBUG)
+        set(${BLOSC_EXTERNAL_LIB}_LIBRARY_DEBUG ${${BLOSC_EXTERNAL_LIB}_LIBRARY_RELEASE})
+        set(${BLOSC_EXTERNAL_LIB}_LIBRARY       ${${BLOSC_EXTERNAL_LIB}_LIBRARY_RELEASE})
+        set(${BLOSC_EXTERNAL_LIB}_LIBRARIES     ${${BLOSC_EXTERNAL_LIB}_LIBRARY_RELEASE})
+      endif()
+
+      # if only the debug version was found, set the release variable also to the debug version
+      if(${BLOSC_EXTERNAL_LIB}_LIBRARY_DEBUG AND NOT ${BLOSC_EXTERNAL_LIB}_LIBRARY_RELEASE)
+        set(${BLOSC_EXTERNAL_LIB}_LIBRARY_RELEASE ${${BLOSC_EXTERNAL_LIB}_LIBRARY_DEBUG})
+        set(${BLOSC_EXTERNAL_LIB}_LIBRARY         ${${BLOSC_EXTERNAL_LIB}_LIBRARY_DEBUG})
+        set(${BLOSC_EXTERNAL_LIB}_LIBRARIES       ${${BLOSC_EXTERNAL_LIB}_LIBRARY_DEBUG})
+      endif()
+
+      target_link_libraries(Blosc::blosc INTERFACE
+        $<$<CONFIG:Release>:${${BLOSC_EXTERNAL_LIB}_LIBRARY_RELEASE}>
+        $<$<CONFIG:Debug>:${${BLOSC_EXTERNAL_LIB}_LIBRARY_DEBUG}>)
+
+    endforeach()
+  endif()
+
+endif()
diff --git a/cmake/modules/FindOpenVDB.cmake b/cmake/modules/FindOpenVDB.cmake
index a091002347..05c8eff904 100644
--- a/cmake/modules/FindOpenVDB.cmake
+++ b/cmake/modules/FindOpenVDB.cmake
@@ -348,6 +348,9 @@ macro(just_fail msg)
   return()
 endmacro()
 
+if (OpenEXR_VERSION VERSION_LESS "3.0.0")
+# only older version of library have this library.
+# for new library - need to detect another librarys (see changelog on OpenEXR)
 find_package(IlmBase QUIET)
 if(NOT IlmBase_FOUND)
   pkg_check_modules(IlmBase QUIET IlmBase)
@@ -367,6 +370,10 @@ if (IlmBase_FOUND AND NOT TARGET IlmBase::Half)
 elseif(NOT IlmBase_FOUND)
   just_fail("IlmBase::Half can not be found!")
 endif()
+else()
+  #for OpenEXR 3.0.0 and later, IlmBase is no longer a dependency, but the IlmThread, Iex and Imath libraries are still required
+  find_package(Imath REQUIRED)
+endif()
 find_package(TBB ${_quiet} ${_required} COMPONENTS tbb)
 find_package(ZLIB ${_quiet} ${_required})
 find_package(Boost ${_quiet} ${_required} COMPONENTS iostreams system )
@@ -469,11 +476,16 @@ endif()
 # include paths from shared installs (including houdini) may pull in the wrong
 # headers
 
-set(_OPENVDB_VISIBLE_DEPENDENCIES
-  Boost::iostreams
-  Boost::system
-  IlmBase::Half
-)
+set(_OPENVDB_VISIBLE_DEPENDENCIES Boost::iostreams)
+if (OpenEXR_VERSION VERSION_LESS "3.0.0")
+    list(APPEND _OPENVDB_VISIBLE_DEPENDENCIES IlmBase::Half)
+else()
+    list(APPEND _OPENVDB_VISIBLE_DEPENDENCIES Imath::Imath)
+endif()
+if (NOT SYS_LIB_BOOST)
+    # Have only on static library
+    list(APPEND _OPENVDB_VISIBLE_DEPENDENCIES Boost::system)
+endif()
 
 set(_OPENVDB_DEFINITIONS)
 if(OpenVDB_ABI)
diff --git a/deps/Boost/Boost.cmake b/deps/Boost/Boost.cmake
index bdd801857e..d6e1e2b2f8 100644
--- a/deps/Boost/Boost.cmake
+++ b/deps/Boost/Boost.cmake
@@ -26,8 +26,8 @@ endif ()
 
 orcaslicer_add_cmake_project(Boost
     ${_options}
-    URL "https://github.com/boostorg/boost/releases/download/boost-1.84.0/boost-1.84.0.tar.gz"
-    URL_HASH SHA256=4d27e9efed0f6f152dc28db6430b9d3dfb40c0345da7342eaa5a987dde57bd95
+    URL "https://github.com/boostorg/boost/releases/download/boost-1.91.0-1/boost-1.91.0-1-cmake.tar.xz"
+    URL_HASH SHA256=cc5dc5006ecbdf0051f90979be31b4eee5987d9ae14ae9fb9c03cfa43fa3cdad
     LIST_SEPARATOR |
     CMAKE_ARGS
         -DBOOST_EXCLUDE_LIBRARIES:STRING=contract|fiber|numpy|stacktrace|wave|test
diff --git a/deps/CMakeLists.txt b/deps/CMakeLists.txt
index 26a1860e33..f8e0bf1333 100644
--- a/deps/CMakeLists.txt
+++ b/deps/CMakeLists.txt
@@ -111,6 +111,44 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     option(DEP_WX_GTK3 "Build wxWidgets against GTK3" ON)
 endif()
 
+if(NOT DEFINED ENV{SYS_LIBS})
+    set(SYS_LIBS "")
+endif()
+option(SYS_LIBS "Compile OrcaSlicer with system libraries, if possible (e.g. 'png;curl')" ${SYS_LIBS})
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/../cmake/modules/)
+
+if (SYS_LIBS)
+    #TODO: boost;cgal
+    set(SYS_LIBS_ALL "zlib;png;gmp;mpfr;openssl;curl;jpeg;freetype;expat;opencv;cereal;qhull;opencsg;tbb;glew;glfw;nlopt;blosc;occt;openexr;openvdb;wxwidgets;boost")
+    if (SYS_LIBS MATCHES "all")
+        set(SYS_LIBS ${SYS_LIBS_ALL})
+    endif()
+    FOREACH(lib ${SYS_LIBS})
+        if (NOT lib IN_LIST SYS_LIBS_ALL)
+            message(FATAL_ERROR "Unknown library specified in SYS_LIBS: ${lib}. Valid options are: ${SYS_LIBS_ALL} or 'all'.")
+        endif()
+        string(TOUPPER ${lib} lib_upper)
+        set(SYS_LIB_${lib_upper} ON CACHE BOOL "Try to use system library for ${lib}")
+        message(STATUS "System library for ${lib}: ${SYS_LIB_${lib_upper}}")
+    ENDFOREACH()
+    if ("zlib" IN_LIST SYS_LIBS)
+        if (NOT "openssl" IN_LIST SYS_LIBS)
+            message(FATAL_ERROR "Using system zlib without system OpenSSL might cause build issues.")
+        endif()
+    endif()
+    #if (("glfw" IN_LIST SYS_LIBS) OR ("glew" IN_LIST SYS_LIBS))
+    #    if (NOT "openvdb" IN_LIST SYS_LIBS)
+    #        message(FATAL_ERROR "glew and glfw requires openvdb to be used as a system library as well. Please add 'openvdb' to SYS_LIBS.")
+    #    endif()
+    #endif()
+    if ("openexr" IN_LIST SYS_LIBS)
+        if (NOT "openvdb" IN_LIST SYS_LIBS)
+            message(FATAL_ERROR "Using system OpenEXR without system OpenVDB might cause build issues.")
+        endif()
+    endif()
+endif()
+
 set(IS_CROSS_COMPILE FALSE)
 
 if (APPLE)
@@ -318,51 +356,176 @@ if(FLATPAK)
     find_package(EXPAT)
     find_package(CURL)
     find_package(JPEG)
-    find_package(Freetype)
     find_package(OpenSSL 1.1...<3.2)
-    find_package(CURL)
 endif()
 
 set(ZLIB_PKG "")
-if (NOT ZLIB_FOUND)
+find_package(ZLIB QUIET)
+if (ZLIB_FOUND AND SYS_LIB_ZLIB)
+    message(STATUS "Using ZLIB from system.")
+    add_library(dep_ZLIB INTERFACE)
+else()
+    message(STATUS "Using ZLIB from source.")
     include(ZLIB/ZLIB.cmake)
     set(ZLIB_PKG dep_ZLIB)
-endif ()
+endif()
+
 set(PNG_PKG "")
-if (NOT PNG_FOUND)
+find_package(PNG QUIET)
+if (PNG_FOUND AND SYS_LIB_PNG)
+    message(STATUS "Using PNG from system.")
+    add_library(dep_PNG INTERFACE)
+else()
+    message(STATUS "Using PNG from source.")
     include(PNG/PNG.cmake)
     set(PNG_PKG dep_PNG)
-endif ()
+endif()
+
 set(EXPAT_PKG "")
-find_package(EXPAT)
-if (NOT EXPAT_FOUND)
+find_package(EXPAT QUIET)
+if (EXPAT_FOUND AND SYS_LIB_EXPAT)
+    message(STATUS "Using EXPAT from system.")
+    add_library(dep_EXPAT INTERFACE)
+else()
+    message(STATUS "Using EXPAT from source.")
     include(EXPAT/EXPAT.cmake)
     set(EXPAT_PKG dep_EXPAT)
-endif ()
+endif()
+
+set(Boost_COMPONENTS filesystem thread log log_setup locale regex chrono atomic date_time iostreams program_options nowide process)
+if(NOT SYS_LIB_BOOST)
+    list(APPEND Boost_COMPONENTS system)
+endif()
+find_package(Boost 1.90.0 QUIET COMPONENTS ${Boost_COMPONENTS})
+if(Boost_FOUND AND SYS_LIB_BOOST)
+    message(STATUS "Using Boost from system.")
+    add_library(dep_Boost INTERFACE)
+else()
+    message(STATUS "Using Boost from source.")
+    set(DEP_Boost_COMPONENTS system iostreams filesystem thread log locale regex date_time)
+    include(Boost/Boost.cmake)
+endif()
 
 # The order of includes respects the dependencies between libraries
-set(DEP_Boost_COMPONENTS system iostreams filesystem thread log locale regex date_time)
-include(Boost/Boost.cmake)
+find_package(cereal QUIET)
+if (cereal_FOUND AND SYS_LIB_CEREAL)
+    message(STATUS "Using cereal from system.")
+    add_library(dep_Cereal INTERFACE)
+else()
+    message(STATUS "Using cereal from source.")
+    include(Cereal/Cereal.cmake)
+endif()
 
-include(Cereal/Cereal.cmake)
-include(Qhull/Qhull.cmake)
-include(GLEW/GLEW.cmake)
+find_package(Qhull QUIET)
+if (Qhull_FOUND AND SYS_LIB_QHULL)
+    message(STATUS "Using Qhull from system.")
+    add_library(dep_Qhull INTERFACE)
+else()
+    message(STATUS "Using Qhull from source.")
+    include(Qhull/Qhull.cmake)
+endif()
 
-include(GLFW/GLFW.cmake)
-include(OpenCSG/OpenCSG.cmake)
+find_package(GLEW QUIET)
+if (GLEW_FOUND AND SYS_LIB_GLEW)
+    message(STATUS "Using GLEW from system.")
+    add_library(dep_GLEW INTERFACE)
+else()
+    message(STATUS "Using GLEW from source.")
+    include(GLEW/GLEW.cmake)
+endif()
 
-include(TBB/TBB.cmake)
+find_package(glfw3 QUIET)
+if(glfw3_FOUND AND SYS_LIB_GLFW)
+    message(STATUS "Using GLFW from system.")
+    add_library(dep_GLFW INTERFACE)
+else()
+    message(STATUS "Using GLFW from source.")
+    include(GLFW/GLFW.cmake)
+endif()
 
-include(Blosc/Blosc.cmake)
-include(OpenEXR/OpenEXR.cmake)
-include(OpenVDB/OpenVDB.cmake)
+find_library(OpenCSG_LIBRARY NAMES opencsg libopencsg QUIET)
+if (OpenCSG_LIBRARY AND SYS_LIB_OPENCSG)
+    message(STATUS "Using OpenCSG from system.")
+    add_library(dep_OpenCSG INTERFACE)
+else()
+    message(STATUS "Using OpenCSG from source.")
+    include(OpenCSG/OpenCSG.cmake)
+endif()
 
-include(GMP/GMP.cmake)
-include(MPFR/MPFR.cmake)
-include(Eigen/Eigen.cmake)
+find_package(TBB QUIET)
+if(TBB_FOUND AND SYS_LIB_TBB)
+    message(STATUS "Using TBB from system.")
+    add_library(dep_TBB INTERFACE)
+else()
+    message(STATUS "Using TBB from source.")
+    include(TBB/TBB.cmake)
+endif()
+
+find_package(Blosc QUIET)
+if(Blosc_FOUND AND SYS_LIB_BLOSC)
+    message(STATUS "Using Blosc from system.")
+    add_library(dep_Blosc INTERFACE)
+else()
+    message(STATUS "Using Blosc from source.")
+    include(Blosc/Blosc.cmake)
+endif()
+
+find_package(OpenEXR QUIET)
+if(OpenEXR_FOUND AND SYS_LIB_OPENEXR)
+    message(STATUS "Using OpenEXR from system.")
+    add_library(dep_OpenEXR INTERFACE)
+else()
+    message(STATUS "Using OpenEXR from source.")
+    include(OpenEXR/OpenEXR.cmake)
+endif()
+
+find_package(OpenVDB 5.0 COMPONENTS openvdb QUIET)
+if(OpenVDB_FOUND AND SYS_LIB_OPENVDB)
+    message(STATUS "Using OpenVDB from system.")
+    add_library(dep_OpenVDB INTERFACE)
+else()
+    message(STATUS "Using OpenVDB from source.")
+    include(OpenVDB/OpenVDB.cmake)
+endif()
+
+find_library(GMP_LIBRARY NAMES gmp libgmp QUIET)
+if(GMP_LIBRARY AND SYS_LIB_GMP)
+    message(STATUS "Using GMP from system.")
+    add_library(dep_GMP INTERFACE)
+else()
+    message(STATUS "Using GMP from source.")
+    include(GMP/GMP.cmake)
+endif()
+
+find_library(MPFR_LIBRARY NAMES mpfr libmpfr QUIET)
+if(MPFR_LIBRARY AND SYS_LIB_MPFR)
+    message(STATUS "Using MPFR from system.")
+    add_library(dep_MPFR INTERFACE)
+else()
+    message(STATUS "Using MPFR from source.")
+    include(MPFR/MPFR.cmake)
+endif()
+
+find_package(Eigen3 5.0.1 QUIET)
+message(STATUS "Eigen3 version found: ${EIGEN3_VERSION}")
+if(Eigen3_FOUND AND SYS_LIB_EIGEN)
+    message(STATUS "Using Eigen from system.")
+    add_library(dep_Eigen INTERFACE)
+else()
+    message(STATUS "Using Eigen from source.")
+    include(Eigen/Eigen.cmake)
+endif()
 include(CGAL/CGAL.cmake)
 
-include(NLopt/NLopt.cmake)
+find_package(NLopt QUIET)
+if(NLopt_FOUND AND SYS_LIB_NLOPT)
+    message(STATUS "Using NLopt from system.")
+    add_library(dep_NLopt INTERFACE)
+else()
+    message(STATUS "Using NLopt from source.")
+    include(NLopt/NLopt.cmake)
+endif()
+
 include(libnoise/libnoise.cmake)
 
 include(Draco/Draco.cmake)
@@ -373,7 +536,12 @@ include(Draco/Draco.cmake)
 # a grep across the repo shows it is used for other things
 # TODO: update openssl and everything that uses <openssl/md5.h>
 set(OPENSSL_PKG "")
-if(NOT OPENSSL_FOUND)
+find_package(OpenSSL QUIET)
+if(OPENSSL_FOUND AND SYS_LIB_OPENSSL)
+    message(STATUS "Using OpenSSL from system.")
+    add_library(dep_OpenSSL INTERFACE)
+else()
+    message(STATUS "Using OpenSSL from source.")
     include(OpenSSL/OpenSSL.cmake)
     set(OPENSSL_PKG dep_OpenSSL)
 endif()
@@ -382,26 +550,49 @@ endif()
 # so, just don't even bother
 # ...i think this is how it works? change if wrong
 set(CURL_PKG "")
-if (NOT OPENSSL_FOUND OR NOT CURL_FOUND)
+find_package(CURL QUIET)
+if(CURL_FOUND AND SYS_LIB_CURL AND NOT OPENSSL_PKG) # if we're using a custom OpenSSL, we need to build Curl as well to link against it
+    message(STATUS "Using CURL from system.")
+    add_library(dep_CURL INTERFACE)
+else()
+    message(STATUS "Using CURL from source.")
     include(CURL/CURL.cmake)
     set(CURL_PKG dep_CURL)
-endif ()
+endif()
 
 set(JPEG_PKG "")
-if (NOT JPEG_FOUND)
+find_package(JPEG QUIET)
+if(JPEG_FOUND AND SYS_LIB_JPEG)
+    message(STATUS "Using JPEG from system.")
+    add_library(dep_JPEG INTERFACE)
+else()
+    message(STATUS "Using JPEG from source.")
     include(JPEG/JPEG.cmake)
     set(JPEG_PKG dep_JPEG)
 endif()
 
 # flatpak builds wxwidgets separately, so it is not included in the deps target
 set(WXWIDGETS_PKG "")
-include(wxWidgets/wxWidgets.cmake)
-if (NOT FLATPAK)
+find_package(wxWidgets 3.2 QUIET)
+if(wxWidgets_FOUND AND SYS_LIB_WXWIDGETS)
+    message(STATUS "Using wxWidgets from system.")
+    add_library(dep_wxWidgets INTERFACE)
     set(WXWIDGETS_PKG "dep_wxWidgets")
+else()
+    message(STATUS "Using wxWidgets from source.")
+    include(wxWidgets/wxWidgets.cmake)
+    if (NOT FLATPAK)
+        set(WXWIDGETS_PKG "dep_wxWidgets")
+    endif()
 endif()
 
 set(FREETYPE_PKG "")
-if(NOT FREETYPE_FOUND)
+find_package(Freetype QUIET)
+if(Freetype_FOUND AND SYS_LIB_FREETYPE)
+    message(STATUS "Using FreeType from system.")
+    add_library(dep_FREETYPE INTERFACE)
+else()
+    message(STATUS "Using FreeType from source.")
     include(FREETYPE/FREETYPE.cmake)
     set(FREETYPE_PKG "dep_FREETYPE")
 endif()
@@ -420,8 +611,24 @@ if((REV_PARSE_RESULT EQUAL 0) AND (REV_PARSE_OUTPUT STREQUAL "true"))
     file(RELATIVE_PATH BINARY_DIR_REL  ${CMAKE_SOURCE_DIR}/.. ${CMAKE_BINARY_DIR})
 endif ()
 
-include(OCCT/OCCT.cmake)
-include(OpenCV/OpenCV.cmake)
+find_package(OpenCASCADE QUIET)
+if(OpenCASCADE_FOUND AND SYS_LIB_OCCT)
+    message(STATUS "Using OpenCASCADE from system.")
+    add_library(dep_OCCT INTERFACE)
+else()
+    message(STATUS "Using OpenCASCADE from source.")
+    include(OCCT/OCCT.cmake)
+endif()
+
+find_package(OpenCV COMPONENTS core imgproc imgcodecs QUIET)
+message(STATUS "OpenCV version found: ${OpenCV_LIB_COMPONENTS}")
+if(OpenCV_FOUND AND SYS_LIB_OPENCV)
+    message(STATUS "Using OpenCV from system.")
+    add_library(dep_OpenCV INTERFACE)
+else()
+    message(STATUS "Using OpenCV from source.")
+    include(OpenCV/OpenCV.cmake)
+endif()
 
 set(_dep_list
     dep_Boost
diff --git a/deps_src/CMakeLists.txt b/deps_src/CMakeLists.txt
index a4d71c6f27..80d773fa29 100644
--- a/deps_src/CMakeLists.txt
+++ b/deps_src/CMakeLists.txt
@@ -2,31 +2,77 @@ cmake_minimum_required(VERSION 3.13)
 project(deps_src)
 
 # Include all dependency subdirectories
+set(USE_SYS_LIBS ON)  # Flag to indicate whether to use system libraries
 
 # Header-only libraries (INTERFACE)
 add_subdirectory(agg)
 add_subdirectory(ankerl)
 add_subdirectory(earcut)
-add_subdirectory(fast_float)
-add_subdirectory(nanosvg)
-add_subdirectory(nlohmann)
+
+find_package(FastFloat QUIET)
+if (USE_SYS_LIBS AND FastFloat_FOUND)
+    message(STATUS "Using system FastFloat library")
+else()
+    add_subdirectory(fast_float)
+endif()
+
+find_package(NanoSVG QUIET)
+if (USE_SYS_LIBS AND NanoSVG_FOUND)
+    message(STATUS "Using system NanoSVG library")
+else()
+    add_subdirectory(nanosvg)
+endif()
+
+find_package(nlohmann_json QUIET)
+if (USE_SYS_LIBS AND nlohmann_json_FOUND)
+    message(STATUS "Using system nlohmann_json library")
+else()
+    add_subdirectory(nlohmann)
+endif()
+
 add_subdirectory(stb_dxt)     # Header-only STB DXT compression library
 
 # Static libraries
 add_subdirectory(Shiny)
 add_subdirectory(admesh)
 add_subdirectory(clipper)
-add_subdirectory(clipper2)
-add_subdirectory(expat)
+
+find_package(Clipper2 QUIET)
+if (USE_SYS_LIBS AND Clipper2_FOUND)
+    message(STATUS "Using system Clipper2 library")
+else()
+    add_subdirectory(clipper2)
+endif()
+
+find_package(expat QUIET)
+if (USE_SYS_LIBS AND expat_FOUND)
+    message(STATUS "Using system expat library")
+else()
+    add_subdirectory(expat)
+endif()
+
 add_subdirectory(glu-libtess)
-add_subdirectory(hidapi)
+
+find_package(hodapi QUIET)
+if (USE_SYS_LIBS AND hidapi_FOUND)
+    message(STATUS "Using system hidapi library")
+else()
+    add_subdirectory(hidapi)
+endif()
+
 add_subdirectory(hints)        # Hints library with utility executable
 add_subdirectory(imgui)
 add_subdirectory(imguizmo)
 add_subdirectory(libigl)
 add_subdirectory(libnest2d)
 add_subdirectory(mcut)
-add_subdirectory(md4c)
+
+find_package(md4c QUIET)
+if (USE_SYS_LIBS AND md4c_FOUND)
+    message(STATUS "Using system md4c library")
+else()
+    add_subdirectory(md4c)
+endif()
 add_subdirectory(mdns)
 add_subdirectory(miniz)
 add_subdirectory(minilzo)
diff --git a/scripts/flatpak/com.orcaslicer.OrcaSlicer.yml b/scripts/flatpak/com.orcaslicer.OrcaSlicer.yml
index ef4ff42c60..c9dbc65f4f 100644
--- a/scripts/flatpak/com.orcaslicer.OrcaSlicer.yml
+++ b/scripts/flatpak/com.orcaslicer.OrcaSlicer.yml
@@ -176,8 +176,8 @@ modules:
 
       # Boost 1.84.0
       - type: file
-        url: https://github.com/boostorg/boost/releases/download/boost-1.84.0/boost-1.84.0.tar.gz
-        sha256: 4d27e9efed0f6f152dc28db6430b9d3dfb40c0345da7342eaa5a987dde57bd95
+        url: https://github.com/boostorg/boost/releases/download/boost-1.91.0-1/boost-1.91.0-1-cmake.tar.xz
+        sha256: cc5dc5006ecbdf0051f90979be31b4eee5987d9ae14ae9fb9c03cfa43fa3cdad
         dest: external-packages/Boost
 
       # Eigen 5.0.1
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 73c767dad2..ba29ec018d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -22,7 +22,7 @@ if (SLIC3R_GUI)
         message(STATUS "WXWIN environment set to: $ENV{WXWIN}")
     elseif(UNIX)
         set(wxWidgets_USE_UNICODE ON)
-        if(SLIC3R_STATIC)
+        if(SLIC3R_STATIC AND NOT SYS_LIB_WXWIDGETS)
             set(wxWidgets_USE_STATIC ON)
         else()
             set(wxWidgets_USE_STATIC OFF)
diff --git a/src/libslic3r/CMakeLists.txt b/src/libslic3r/CMakeLists.txt
index 9b67d7ab50..eea68c983a 100644
--- a/src/libslic3r/CMakeLists.txt
+++ b/src/libslic3r/CMakeLists.txt
@@ -489,7 +489,7 @@ add_library(libslic3r STATIC ${lisbslic3r_sources}
     ${OpenVDBUtils_SOURCES})
 source_group(TREE ${CMAKE_CURRENT_SOURCE_DIR} FILES ${lisbslic3r_sources})
 
-if (SLIC3R_STATIC)
+if (SLIC3R_STATIC AND NOT SYS_LIB_CGAL)
     set(CGAL_Boost_USE_STATIC_LIBS ON CACHE BOOL "" FORCE)
 endif ()
 set(CGAL_DO_NOT_WARN_ABOUT_CMAKE_BUILD_TYPE ON CACHE BOOL "" FORCE)
@@ -497,8 +497,12 @@ set(CGAL_DO_NOT_WARN_ABOUT_CMAKE_BUILD_TYPE ON CACHE BOOL "" FORCE)
 cmake_policy(PUSH)
 cmake_policy(SET CMP0011 NEW)
 find_package(CGAL REQUIRED)
-find_package(OpenCV REQUIRED core)
 cmake_policy(POP)
+find_package(OpenCV REQUIRED core imgproc imgcodecs)
+if (SLIC3R_STATIC AND NOT SYS_LIB_OPENCV)
+    # fix-up for a static linked version
+    set(OpenCV_LIBS opencv_world)
+endif()
 
 add_library(libslic3r_cgal STATIC
     CutSurface.hpp CutSurface.cpp
@@ -545,12 +549,21 @@ target_include_directories(libslic3r SYSTEM PUBLIC ${OpenCASCADE_INCLUDE_DIR})
 find_package(JPEG REQUIRED)
 find_package(draco REQUIRED)
 
-set(OCCT_LIBS
-    TKXDESTEP
-    TKSTEP
-    TKSTEP209
-    TKSTEPAttr
-    TKSTEPBase
+if (OpenCASCADE_VERSION VERSION_LESS 7.8.0)
+    set(OCCT_LIBS
+        TKXDESTEP
+        TKSTEP
+        TKSTEP209
+        TKSTEPAttr
+        TKSTEPBase)
+else()
+    # OCCT 7.8.0 and later use another set of libraries
+    set(OCCT_LIBS
+        TKXSDRAWSTEP
+        TKDESTEP)
+endif()
+
+list(APPEND OCCT_LIBS
     TKXCAF
     TKXSBase
     TKVCAF
@@ -581,7 +594,7 @@ target_link_libraries(libslic3r
         libigl
         libnest2d
         miniz
-        opencv_world
+        ${OpenCV_LIBS}
     PRIVATE
         ${CMAKE_DL_LIBS}
         ${EXPAT_LIBRARIES}
diff --git a/src/libslic3r/GCode/PostProcessor.cpp b/src/libslic3r/GCode/PostProcessor.cpp
index 1e084cd35e..732ce711b1 100644
--- a/src/libslic3r/GCode/PostProcessor.cpp
+++ b/src/libslic3r/GCode/PostProcessor.cpp
@@ -11,6 +11,12 @@
 #include <boost/nowide/cstdlib.hpp>
 #include <boost/nowide/convert.hpp>
 #include <boost/nowide/fstream.hpp>
+#include <boost/process/v1/child.hpp>
+#include <boost/process/v1/pipe.hpp>
+#include <boost/process/v1/start_dir.hpp>
+#include <boost/process/v1/io.hpp>
+#include <boost/process/v1/handles.hpp>
+#include <boost/asio.hpp>
 
 // BBS
 #include <iostream>
@@ -170,8 +176,8 @@ static int run_script(const std::string &script, const std::string &gcode, std::
 
     BOOST_LOG_TRIVIAL(debug) << boost::format("Executing script, shell: %1%, command: %2%") % shell % command;
 
-    process::ipstream istd_err;
-    process::child child(shell, "-c", command, process::std_err > istd_err);
+    process::v1::ipstream istd_err;
+    process::v1::child child(shell, "-c", command, process::v1::std_err > istd_err);
 
     std_err.clear();
     std::string line;
diff --git a/src/slic3r/CMakeLists.txt b/src/slic3r/CMakeLists.txt
index 969dcb693a..792d8a7959 100644
--- a/src/slic3r/CMakeLists.txt
+++ b/src/slic3r/CMakeLists.txt
@@ -816,7 +816,7 @@ if (UNIX AND NOT APPLE)
     find_package(GTK${SLIC3R_GTK} REQUIRED)
     pkg_check_modules(LIBSECRET REQUIRED libsecret-1)
     pkg_check_modules(webkit2gtk REQUIRED webkit2gtk-4.1)
-    if (FLATPAK)
+    if (FLATPAK OR SYS_LIB_WXWIDGETS)
         # I don't know why this is needed, but for whatever reason slic3r isn't
         # linking to X11 and webkit2gtk. force it.
         find_package(X11 REQUIRED)
diff --git a/src/slic3r/Config/Version.cpp b/src/slic3r/Config/Version.cpp
index 092594ce51..f53fbbea2c 100644
--- a/src/slic3r/Config/Version.cpp
+++ b/src/slic3r/Config/Version.cpp
@@ -2,7 +2,7 @@
 
 #include <cctype>
 
-#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/directory.hpp>
 #include <boost/nowide/fstream.hpp>
 
 #include "libslic3r/libslic3r.h"
diff --git a/src/slic3r/GUI/GUI_App.cpp b/src/slic3r/GUI/GUI_App.cpp
index b6ab6dc706..83201bcd3e 100644
--- a/src/slic3r/GUI/GUI_App.cpp
+++ b/src/slic3r/GUI/GUI_App.cpp
@@ -8398,7 +8398,7 @@ void GUI_App::preset_deleted_from_cloud(std::string setting_id)
 
     // Delete the .info file after cloud deletion is confirmed
     if (!preset_file_path.empty() && fs::exists(fs::path(preset_file_path))) {
-        boost::nowide::remove(preset_file_path.c_str());
+        boost::filesystem::remove(preset_file_path.c_str());
         BOOST_LOG_TRIVIAL(info) << "Deleted .info file after cloud confirmation: " << preset_file_path;
     }
 
diff --git a/src/slic3r/GUI/HttpServer.hpp b/src/slic3r/GUI/HttpServer.hpp
index 4c0b929663..0976092318 100644
--- a/src/slic3r/GUI/HttpServer.hpp
+++ b/src/slic3r/GUI/HttpServer.hpp
@@ -130,7 +130,7 @@ private:
     {
     public:
         HttpServer&                        server;
-        boost::asio::io_service            io_service;
+        boost::asio::io_context            io_service;
         boost::asio::ip::tcp::acceptor     acceptor;
         std::set<std::shared_ptr<session>> sessions;
 
diff --git a/src/slic3r/GUI/MediaPlayCtrl.cpp b/src/slic3r/GUI/MediaPlayCtrl.cpp
index 5737484960..5429f287c8 100644
--- a/src/slic3r/GUI/MediaPlayCtrl.cpp
+++ b/src/slic3r/GUI/MediaPlayCtrl.cpp
@@ -16,6 +16,12 @@
 #include <boost/nowide/utf8_codecvt.hpp>
 #undef pid_t
 #include <boost/process.hpp>
+#include <boost/process/v1/child.hpp>
+#include <boost/process/v1/pipe.hpp>
+#include <boost/process/v1/start_dir.hpp>
+#include <boost/process/v1/io.hpp>
+#include <boost/process/v1/handles.hpp>
+#include <boost/asio.hpp>
 #ifdef __WIN32__
 #include <boost/process/windows.hpp>
 #else
@@ -764,7 +770,7 @@ bool MediaPlayCtrl::start_stream_service(bool *need_install)
         std::vector<std::string> configss;
         boost::algorithm::split(configss, configs, boost::algorithm::is_any_of("\r\n"));
         configss.erase(std::remove(configss.begin(), configss.end(), std::string()), configss.end());
-        boost::process::pipe intermediate;
+        boost::process::v1::pipe intermediate;
         boost::filesystem::path start_dir(boost::filesystem::path(data_dir()) / "plugins");
 #ifdef __WXMSW__
         auto plugins_dir = boost::nowide::widen(data_dir()) + L"\\plugins\\";
@@ -774,17 +780,17 @@ bool MediaPlayCtrl::start_stream_service(bool *need_install)
             if (!boost::filesystem::exists(file_dll) || boost::filesystem::last_write_time(file_dll) != boost::filesystem::last_write_time(file_dll2))
                 boost::filesystem::copy_file(file_dll2, file_dll, boost::filesystem::copy_options::overwrite_existing);
         }
-        boost::process::child process_source(file_source, file_url2.ToStdWstring(), boost::process::start_dir(tools_dir), 
+        boost::process::v1::child process_source(file_source, file_url2.ToStdWstring(), boost::process::start_dir(tools_dir),
                                              boost::process::windows::create_no_window, 
                                              boost::process::std_out > intermediate, boost::process::limit_handles);
-        boost::process::child process_ffmpeg(file_ffmpeg, configss, boost::process::windows::create_no_window, 
+        boost::process::v1::child process_ffmpeg(file_ffmpeg, configss, boost::process::windows::create_no_window,
                                              boost::process::std_in < intermediate, boost::process::limit_handles);
 #else
         boost::filesystem::permissions(file_source, boost::filesystem::owner_exe | boost::filesystem::add_perms);
         boost::filesystem::permissions(file_ffmpeg, boost::filesystem::owner_exe | boost::filesystem::add_perms);
-        boost::process::child process_source(file_source, file_url2.data().AsInternal(), boost::process::start_dir(start_dir), 
-                                             boost::process::std_out > intermediate, boost::process::limit_handles);
-        boost::process::child process_ffmpeg(file_ffmpeg, configss, boost::process::std_in < intermediate, boost::process::limit_handles);
+        boost::process::v1::child process_source(file_source, file_url2.data().AsInternal(), boost::process::v1::start_dir(start_dir),
+                                             boost::process::v1::std_out > intermediate, boost::process::v1::limit_handles);
+        boost::process::v1::child process_ffmpeg(file_ffmpeg, configss, boost::process::v1::std_in < intermediate, boost::process::v1::limit_handles);
 #endif
         process_source.detach();
         process_ffmpeg.detach();
diff --git a/src/slic3r/GUI/RemovableDriveManager.cpp b/src/slic3r/GUI/RemovableDriveManager.cpp
index 83ec4695c7..bca6b7b049 100644
--- a/src/slic3r/GUI/RemovableDriveManager.cpp
+++ b/src/slic3r/GUI/RemovableDriveManager.cpp
@@ -22,7 +22,9 @@
 #include <pwd.h>
 #include <boost/filesystem.hpp>
 #include <boost/system/error_code.hpp>
-#include <boost/process.hpp>
+#include <boost/process/v1/child.hpp>
+#include <boost/process/v1/search_path.hpp>
+#include <boost/process/v1/io.hpp>
 #endif
 
 namespace Slic3r {
@@ -302,15 +304,15 @@ void RemovableDriveManager::eject_drive()
 		// but neither triggers "succesful safe removal messege"
 		
 		BOOST_LOG_TRIVIAL(info) << "Ejecting started";
-		boost::process::ipstream istd_err;
-    	boost::process::child child(
+		boost::process::v1::ipstream istd_err;
+		boost::process::v1::child child(
 #if __APPLE__		
-			boost::process::search_path("diskutil"), "eject", correct_path.c_str(), (boost::process::std_out & boost::process::std_err) > istd_err);
+			boost::process::v1::search_path("diskutil"), "eject", correct_path.c_str(), (boost::process::v1::std_out & boost::process::v1::std_err) > istd_err);
 		//Another option how to eject at mac. Currently not working.
 		//used insted of system() command;
 		//this->eject_device(correct_path);
 #else
-    		boost::process::search_path("umount"), correct_path.c_str(), (boost::process::std_out & boost::process::std_err) > istd_err);
+		boost::process::v1::search_path("umount"), correct_path.c_str(), (boost::process::v1::std_out & boost::process::v1::std_err) > istd_err);
 #endif
 		std::string line;
 		while (child.running() && std::getline(istd_err, line)) {
diff --git a/src/slic3r/GUI/WebUserLoginDialog.cpp b/src/slic3r/GUI/WebUserLoginDialog.cpp
index 4b8acc9455..edc2fa4381 100644
--- a/src/slic3r/GUI/WebUserLoginDialog.cpp
+++ b/src/slic3r/GUI/WebUserLoginDialog.cpp
@@ -48,7 +48,7 @@ namespace {
 int reserve_loopback_port()
 {
     try {
-        boost::asio::io_service       io_service;
+        boost::asio::io_context       io_service;
         boost::asio::ip::tcp::acceptor acceptor(io_service, {boost::asio::ip::tcp::v4(), 0});
         return static_cast<int>(acceptor.local_endpoint().port());
     } catch (...) {
diff --git a/src/slic3r/Utils/Bonjour.cpp b/src/slic3r/Utils/Bonjour.cpp
index 0701695eff..0dce662a0c 100644
--- a/src/slic3r/Utils/Bonjour.cpp
+++ b/src/slic3r/Utils/Bonjour.cpp
@@ -620,7 +620,7 @@ UdpSession::UdpSession(Bonjour::ReplyFn rfn) : replyfn(rfn)
 	buffer.resize(DnsMessage::MAX_SIZE);
 }
 
-UdpSocket::UdpSocket( Bonjour::ReplyFn replyfn, const asio::ip::address& multicast_address, const asio::ip::address& interface_address, std::shared_ptr< boost::asio::io_service > io_service)
+UdpSocket::UdpSocket( Bonjour::ReplyFn replyfn, const asio::ip::address& multicast_address, const asio::ip::address& interface_address, std::shared_ptr< boost::asio::io_context > io_service)
 	: replyfn(replyfn)
 	, multicast_address(multicast_address)
 	, socket(*io_service)
@@ -654,7 +654,7 @@ UdpSocket::UdpSocket( Bonjour::ReplyFn replyfn, const asio::ip::address& multica
 }
 
 
-UdpSocket::UdpSocket( Bonjour::ReplyFn replyfn, const asio::ip::address& multicast_address, std::shared_ptr< boost::asio::io_service > io_service)
+UdpSocket::UdpSocket( Bonjour::ReplyFn replyfn, const asio::ip::address& multicast_address, std::shared_ptr< boost::asio::io_context > io_service)
 	: replyfn(replyfn)
 	, multicast_address(multicast_address)
 	, socket(*io_service)
@@ -710,7 +710,9 @@ void UdpSocket::receive_handler(SharedSession session, const boost::system::erro
 	// let io_service to handle the datagram on session
 	// from boost documentation io_service::post:
 	// The io_service guarantees that the handler will only be called in a thread in which the run(), run_one(), poll() or poll_one() member functions is currently being invoked.
-	io_service->post(boost::bind(&UdpSession::handle_receive, session, error, bytes));
+	boost::asio::post(io_service->get_executor(),
+		boost::bind(&UdpSession::handle_receive, session, error, bytes)
+	);
 	// immediately accept new datagrams
 	async_receive();
 }
@@ -867,7 +869,7 @@ void Bonjour::priv::lookup_perform()
 {
 	service_dn = (boost::format("_%1%._%2%.local") % service % protocol).str();
 
-	std::shared_ptr< boost::asio::io_service > io_service(new boost::asio::io_service);
+	std::shared_ptr< boost::asio::io_context > io_service(new boost::asio::io_context);
 
 	std::vector<LookupSocket*> sockets;
 
@@ -919,7 +921,7 @@ void Bonjour::priv::lookup_perform()
 			socket->send();
 
 		// timer settings
-		asio::deadline_timer timer(*io_service);
+		asio::steady_timer timer(*io_service);
 		retries--;
 		std::function<void(const error_code&)> timer_handler = [&](const error_code& error) {
 			// end 
@@ -932,7 +934,7 @@ void Bonjour::priv::lookup_perform()
 			// restart timer
 			} else {
 				retries--;
-				timer.expires_from_now(boost::posix_time::seconds(timeout));
+				timer.expires_at(timer.expiry() + std::chrono::seconds(timeout));
 				timer.async_wait(timer_handler);
 				// trigger another round of queries
 				for (auto * socket : sockets)
@@ -940,7 +942,7 @@ void Bonjour::priv::lookup_perform()
 			}
 		};
 		// start timer
-		timer.expires_from_now(boost::posix_time::seconds(timeout));
+		timer.expires_at(timer.expiry() + std::chrono::seconds(timeout));
 		timer.async_wait(timer_handler);
 		// start io_service, it will run until it has something to do - so in this case until stop is called in timer
 		io_service->run();
@@ -962,7 +964,7 @@ void Bonjour::priv::resolve_perform()
 			rpls.push_back(reply);
 	};
 
-	std::shared_ptr< boost::asio::io_service > io_service(new boost::asio::io_service);
+	std::shared_ptr< boost::asio::io_context > io_service(new boost::asio::io_context);
 	std::vector<ResolveSocket*> sockets;
 
 	// resolve interfaces - from PR#6646
@@ -1012,7 +1014,7 @@ void Bonjour::priv::resolve_perform()
 			socket->send();
 
 		// timer settings
-		asio::deadline_timer timer(*io_service);
+		asio::steady_timer timer(*io_service);
 		retries--;
 		std::function<void(const error_code&)> timer_handler = [&](const error_code& error) {
 			int replies_count = replies.size();
@@ -1026,7 +1028,7 @@ void Bonjour::priv::resolve_perform()
 			// restart timer
 			} else {
 				retries--;
-				timer.expires_from_now(boost::posix_time::seconds(timeout));
+				timer.expires_at(timer.expiry() + std::chrono::seconds(timeout));
 				timer.async_wait(timer_handler);
 				// trigger another round of queries
 				for (auto * socket : sockets)
@@ -1034,7 +1036,7 @@ void Bonjour::priv::resolve_perform()
 			}
 		};
 		// start timer
-		timer.expires_from_now(boost::posix_time::seconds(timeout));
+		timer.expires_at(timer.expiry() + std::chrono::seconds(timeout));
 		timer.async_wait(timer_handler);
 		// start io_service, it will run until it has something to do - so in this case until stop is called in timer
 		io_service->run();
diff --git a/src/slic3r/Utils/Bonjour.hpp b/src/slic3r/Utils/Bonjour.hpp
index 50b71791fa..f77a7bba74 100644
--- a/src/slic3r/Utils/Bonjour.hpp
+++ b/src/slic3r/Utils/Bonjour.hpp
@@ -151,11 +151,11 @@ public:
 	UdpSocket(Bonjour::ReplyFn replyfn
 		, const boost::asio::ip::address& multicast_address
 		, const boost::asio::ip::address& interface_address
-		, std::shared_ptr< boost::asio::io_service > io_service);
+		, std::shared_ptr< boost::asio::io_context > io_service);
 
 	UdpSocket(Bonjour::ReplyFn replyfn
 		, const boost::asio::ip::address& multicast_address
-		, std::shared_ptr< boost::asio::io_service > io_service);
+		, std::shared_ptr< boost::asio::io_context > io_service);
 
 	void send();
 	void async_receive();
@@ -168,7 +168,7 @@ protected:
 	boost::asio::ip::address					    multicast_address;
 	boost::asio::ip::udp::socket					socket;
 	boost::asio::ip::udp::endpoint					mcast_endpoint;
-	std::shared_ptr< boost::asio::io_service >	io_service;
+	std::shared_ptr< boost::asio::io_context >	io_service;
 	std::vector<BonjourRequest>						requests;
 };
 
@@ -182,7 +182,7 @@ public:
 		, Bonjour::ReplyFn replyfn
 		, const boost::asio::ip::address& multicast_address
 		, const boost::asio::ip::address& interface_address
-		, std::shared_ptr< boost::asio::io_service > io_service)
+		, std::shared_ptr< boost::asio::io_context > io_service)
 		: UdpSocket(replyfn, multicast_address, interface_address, io_service)
 		, txt_keys(txt_keys)
 		, service(service)
@@ -199,7 +199,7 @@ public:
 		, std::string protocol
 		, Bonjour::ReplyFn replyfn
 		, const boost::asio::ip::address& multicast_address
-		, std::shared_ptr< boost::asio::io_service > io_service)
+		, std::shared_ptr< boost::asio::io_context > io_service)
 		: UdpSocket(replyfn, multicast_address, io_service)
 		, txt_keys(txt_keys)
 		, service(service)
@@ -237,7 +237,7 @@ public:
 		, Bonjour::ReplyFn replyfn
 		, const boost::asio::ip::address& multicast_address
 		, const boost::asio::ip::address& interface_address
-		, std::shared_ptr< boost::asio::io_service > io_service)
+		, std::shared_ptr< boost::asio::io_context > io_service)
 		: UdpSocket(replyfn, multicast_address, interface_address, io_service)
 		, hostname(hostname)
 
@@ -249,7 +249,7 @@ public:
 	ResolveSocket(const std::string& hostname
 		, Bonjour::ReplyFn replyfn
 		, const boost::asio::ip::address& multicast_address
-		, std::shared_ptr< boost::asio::io_service > io_service)
+		, std::shared_ptr< boost::asio::io_context > io_service)
 		: UdpSocket(replyfn, multicast_address, io_service)
 		, hostname(hostname)
 
diff --git a/src/slic3r/Utils/Serial.cpp b/src/slic3r/Utils/Serial.cpp
index 4db1acc6b6..6bf2bb0f26 100644
--- a/src/slic3r/Utils/Serial.cpp
+++ b/src/slic3r/Utils/Serial.cpp
@@ -278,11 +278,11 @@ std::vector<std::string> scan_serial_ports()
 namespace asio = boost::asio;
 using boost::system::error_code;
 
-Serial::Serial(asio::io_service& io_service) :
+Serial::Serial(asio::io_context& io_service) :
 	asio::serial_port(io_service)
 {}
 
-Serial::Serial(asio::io_service& io_service, const std::string &name, unsigned baud_rate) :
+Serial::Serial(asio::io_context& io_service, const std::string &name, unsigned baud_rate) :
 	asio::serial_port(io_service, name)
 {
 	set_baud_rate(baud_rate);
diff --git a/src/slic3r/Utils/Serial.hpp b/src/slic3r/Utils/Serial.hpp
index 8bad75b315..3eebc771b8 100644
--- a/src/slic3r/Utils/Serial.hpp
+++ b/src/slic3r/Utils/Serial.hpp
@@ -39,8 +39,8 @@ extern std::vector<SerialPortInfo> 	scan_serial_ports_extended();
 class Serial : public boost::asio::serial_port
 {
 public:
-	Serial(boost::asio::io_service &io_service);
-	Serial(boost::asio::io_service &io_service, const std::string &name, unsigned baud_rate);
+	Serial(boost::asio::io_context &io_service);
+	Serial(boost::asio::io_context &io_service, const std::string &name, unsigned baud_rate);
 	Serial(const Serial &) = delete;
 	Serial &operator=(const Serial &) = delete;
 	~Serial();
diff --git a/src/slic3r/Utils/TCPConsole.cpp b/src/slic3r/Utils/TCPConsole.cpp
index 25802e3908..63fd0a4b5a 100644
--- a/src/slic3r/Utils/TCPConsole.cpp
+++ b/src/slic3r/Utils/TCPConsole.cpp
@@ -170,7 +170,7 @@ bool TCPConsole::run_queue()
 
         auto endpoints = m_resolver.resolve(m_host_name, m_port_name);
 
-        m_socket.async_connect(endpoints->endpoint(),
+        m_socket.async_connect(endpoints.begin()->endpoint(),
             boost::bind(&TCPConsole::handle_connect, this, boost::placeholders::_1)
         );
 
