From 24ac43f2f36c4a646d3dac5d6dbc24d9c52a16a2 Mon Sep 17 00:00:00 2001
From: Andrey Leonchikov <andreil499@gmail.com>
Date: Fri, 24 Apr 2026 20:09:22 +0200
Subject: [PATCH 01/11] Use system GMP/MPFR.

---
 src/libslic3r/CMakeLists.txt             |    3 +
 create mode 100755 build.sh

diff --git a/src/libslic3r/CMakeLists.txt b/src/libslic3r/CMakeLists.txt
index 5f9591452f..8c4fb1981a 100644
--- a/src/libslic3r/CMakeLists.txt
+++ b/src/libslic3r/CMakeLists.txt
@@ -544,6 +544,9 @@ target_include_directories(libslic3r SYSTEM PUBLIC ${OpenCASCADE_INCLUDE_DIR})
 find_package(JPEG REQUIRED)
 find_package(draco REQUIRED)
 
+find_library(GMP_LIBRARY NAMES gmp libgmp)
+find_library(MPFR_LIBRARY NAMES mpfr libmpfr)
+
 set(OCCT_LIBS
     TKXDESTEP
     TKSTEP
-- 
2.53.0

