From 246b4dc34534bc09a34b236c6b5e271d74832785 Mon Sep 17 00:00:00 2001
From: Andrey Leonchikov <andreil499@gmail.com>
Date: Fri, 24 Apr 2026 23:26:44 +0200
Subject: [PATCH 02/11] Move to external OCCT library.

---
 build.sh                          |  13 +-
 src/libslic3r/CMakeLists.txt      |  11 +-

diff --git a/src/libslic3r/CMakeLists.txt b/src/libslic3r/CMakeLists.txt
index 8c4fb1981a..4a5b647f53 100644
--- a/src/libslic3r/CMakeLists.txt
+++ b/src/libslic3r/CMakeLists.txt
@@ -537,7 +537,7 @@ target_include_directories(libslic3r PRIVATE ${CMAKE_CURRENT_SOURCE_DIR} PUBLIC
 target_include_directories(libslic3r SYSTEM PUBLIC ${EXPAT_INCLUDE_DIRS})
 
 # Find the OCCT and related libraries
-set(OpenCASCADE_DIR "${CMAKE_PREFIX_PATH}/lib/cmake/occt")
+#set(OpenCASCADE_DIR "${CMAKE_PREFIX_PATH}/lib/cmake/occt")
 find_package(OpenCASCADE REQUIRED)
 target_include_directories(libslic3r SYSTEM PUBLIC ${OpenCASCADE_INCLUDE_DIR})
 
@@ -548,11 +548,8 @@ find_library(GMP_LIBRARY NAMES gmp libgmp)
 find_library(MPFR_LIBRARY NAMES mpfr libmpfr)
 
 set(OCCT_LIBS
-    TKXDESTEP
-    TKSTEP
-    TKSTEP209
-    TKSTEPAttr
-    TKSTEPBase
+    TKXSDRAWSTEP
+    TKDESTEP
     TKXCAF
     TKXSBase
     TKVCAF
-- 
2.53.0

