From 582205f9c03ede885b0706628ec489f3eb28cb91 Mon Sep 17 00:00:00 2001
From: Andrey Leonchikov <andreil499@gmail.com>
Date: Sat, 25 Apr 2026 22:34:59 +0200
Subject: [PATCH 07/11] Move OpenCV to a system lib.

---
 src/libslic3r/CMakeLists.txt         |  5 +-

diff --git a/src/libslic3r/CMakeLists.txt b/src/libslic3r/CMakeLists.txt
index 53e4e3a064..6c9dff04c5 100644
--- a/src/libslic3r/CMakeLists.txt
+++ b/src/libslic3r/CMakeLists.txt
@@ -496,7 +496,8 @@ set(CGAL_DO_NOT_WARN_ABOUT_CMAKE_BUILD_TYPE ON CACHE BOOL "" FORCE)
 cmake_policy(PUSH)
 cmake_policy(SET CMP0011 NEW)
 find_package(CGAL REQUIRED)
-find_package(OpenCV REQUIRED core)
+find_package(OpenCV REQUIRED)
+include_directories(${OpenCV_INCLUDE_DIRS})
 cmake_policy(POP)
 
 add_library(libslic3r_cgal STATIC 
@@ -586,6 +587,6 @@ target_link_libraries(libslic3r
         libigl
         libnest2d
         miniz
-        opencv_world
+        ${OpenCV_LIBS}
         GLEW::GLEW
     PRIVATE
-- 
2.53.0

