From 989a9c653d71403f9dcb7f617ccd6a8062fa00d0 Mon Sep 17 00:00:00 2001
From: Andrey Leonchikov <andreil499@gmail.com>
Date: Sat, 25 Apr 2026 23:15:36 +0200
Subject: [PATCH 08/11] Move Wx to system lib.

---
 src/CMakeLists.txt             |  2 +-
 src/slic3r/CMakeLists.txt      |  5 ++

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f7de86410d..90ef4ba3c8 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -31,11 +31,11 @@ if (SLIC3R_GUI)
 
     if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
         set (wxWidgets_CONFIG_OPTIONS "--toolkit=gtk${SLIC3R_GTK}")
         if (SLIC3R_WX_STABLE)
-            find_package(wxWidgets 3.0 REQUIRED COMPONENTS base core adv html gl aui net media webview)
+            find_package(wxWidgets 3.2 REQUIRED COMPONENTS base core adv html gl aui net media webview)
         else ()
-            find_package(wxWidgets 3.1 REQUIRED COMPONENTS base core adv html gl aui net media webview)
+            find_package(wxWidgets 3.2 REQUIRED COMPONENTS base core adv html gl aui net media webview)
         endif ()
     else ()
         find_package(wxWidgets 3.1 CONFIG REQUIRED COMPONENTS html adv gl core base webview aui net media)
     endif ()
diff --git a/src/slic3r/CMakeLists.txt b/src/slic3r/CMakeLists.txt
index 1cd2c149bf..fef7477c6f 100644
--- a/src/slic3r/CMakeLists.txt
+++ b/src/slic3r/CMakeLists.txt
@@ -782,6 +782,11 @@ if (SLIC3R_PCH AND NOT SLIC3R_SYNTAXONLY)
     add_precompiled_header(libslic3r_gui pchheader.hpp FORCEINCLUDE)
 endif ()
 
+find_package(X11 REQUIRED)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(webkit2gtk REQUIRED webkit2gtk-4.1)
+target_link_libraries (libslic3r_gui ${X11_LIBRARIES} ${webkit2gtk_LIBRARIES})
+
 # We need to implement some hacks for wxWidgets and touch the underlying GTK
 # layer and sub-libraries. This forces us to use the include locations and
 # link these libraries.
diff --git a/src/slic3r/GUI/PhysicalPrinterDialog.cpp b/src/slic3r/GUI/PhysicalPrinterDialog.cpp
index 6e2eee84..702115e8 100644
--- a/src/slic3r/GUI/PhysicalPrinterDialog.cpp
+++ b/src/slic3r/GUI/PhysicalPrinterDialog.cpp
@@ -429,7 +429,7 @@ void PhysicalPrinterDialog::build_printhost_settings(ConfigOptionsGroup* m_optgr
     // Always fill in the "printhost_port" combo box from the config and select it.
     {
         Choice* choice = dynamic_cast<Choice*>(m_optgroup->get_field("printhost_port"));
-        choice->set_values({ m_config->opt_string("printhost_port") });
+        choice->set_values(wxArrayString{ m_config->opt_string("printhost_port") });
         choice->set_selection();
     }
 
-- 
2.53.0

